tsdb/engine.go

```diff
@@ -60,6 +74,21 @@ func NewEngine(path string, walPath string, options EngineOptions) (Engine, erro
 	// Only bolt-based backends are currently supported so open it and check the format.
 	var format string
 	if err := func() error {
+		// if it's a dir then it's a tsm1 engine
```

Comment:
Just to be clear, if one upgraded and created a new engine that also used directories in this manner, and then downgraded, this code will not be happy.